tests/blackbox_test.go

```diff
@@ -0,0 +1,1215 @@
+			"--change-name=%d:%s", p.Number, p.Label))
+		if p.TypeGUID != "" {
+			opts = append(opts, fmt.Sprintf(
+				"--typecode=%d:%s", p.Number, p.TypeGUID))
+		}
+		if p.GUID != "" {
+			opts = append(opts, fmt.Sprintf(
+				"--partition-guid=%d:%s", p.Number, p.GUID))
+		}
+		if p.Hybrid {
+			hybrids = append(hybrids, p.Number)
+		}
+	}
+	if len(hybrids) > 0 {
+		if len(hybrids) > 3 {
+			t.Fatal("Can't have more than three hybrids")
+		} else {
+			opts = append(opts, fmt.Sprintf("-h=%s", intJoin(hybrids, ":")))
+		}
+	}
+	sgdiskOut, err := exec.Command(
+		"/sbin/sgdisk", opts...).CombinedOutput()
+	if err != nil {
+		t.Fatal("sgdisk", err, string(sgdiskOut))
+	}
+}
+
+func kpartxAdd(t *testing.T, fileName string) string {
+	kpartxOut, err := exec.Command(
+		"/sbin/kpartx", "-av", fileName).CombinedOutput()
+	if err != nil {
+		t.Fatal("kpartx", err, string(kpartxOut))
+	}
+	kpartxOut, err = exec.Command(
+		"/sbin/kpartx", "-l", fileName).CombinedOutput()
+	return strings.Trim(strings.Split(string(kpartxOut), " ")[4], "/dev/")
+}
+
+func mountRootPartition(t *testing.T, partitions []*Partition) {
+	for _, partition := range partitions {
+		if partition.Label != "ROOT" {
+			continue
+		}
+		mountOut, err := exec.Command(
+			"/bin/mount", partition.Device,
+			partition.MountPath).CombinedOutput()
+		if err != nil {
+			t.Fatal("mount", err, string(mountOut))
+		}
+		return
+	}
+	t.Fatal("Didn't find the ROOT partition to mount")
+}
+
+func mountPartitions(t *testing.T, partitions []*Partition) {
+	for _, partition := range partitions {
+		if partition.FilesystemType == "" || partition.Label == "ROOT" {
+			continue
+		}
+		mountOut, err := exec.Command(
+			"/bin/mount", partition.Device,
+			partition.MountPath).CombinedOutput()
+		if err != nil {
+			t.Fatal("mount", err, string(mountOut))
+		}
+	}
+}
+
+func updateTypeGUID(t *testing.T, partition *Partition) {
+	switch partition.TypeCode {
+	case "coreos-resize":
+		partition.TypeGUID = "3884DD41-8582-4404-B9A8-E9B84F2DF50E"
+	case "data":
+		partition.TypeGUID = "0FC63DAF-8483-4772-8E79-3D69D8477DE4"
+	case "coreos-rootfs":
+		partition.TypeGUID = "5DFBF5F4-2848-4BAC-AA5E-0D9A20B745A6"
+	case "bios":
+		partition.TypeGUID = "21686148-6449-6E6F-744E-656564454649"
+	case "efi":
+		partition.TypeGUID = "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+	case "coreos-reserved":
+		partition.TypeGUID = "C95DC21A-DF0E-4340-8D7B-26CBFA9A03E0"
+	case "", "blank":
+		return
+	default:
+		t.Fatal("Unknown TypeCode", partition.TypeCode)
+	}
+}
+
+func intJoin(ints []int, delimiter string) string {
+	strArr := []string{}
+	for _, i := range ints {
+		strArr = append(strArr, strconv.Itoa(i))
+	}
+	return strings.Join(strArr, delimiter)
+}
+
+func removeEmpty(strings []string) []string {
+	var r []string
+	for _, str := range strings {
+		if str != "" {
+			r = append(r, str)
+		}
+	}
+	return r
+}
+
+func generateUUID(t *testing.T) string {
+	out, err := exec.Command("/usr/bin/uuidgen").CombinedOutput()
```

Comment:
Any reason you're hardcoding the path of all the binaries in this? Why not let `exec` find it in the `$PATH`? On some OSs (*cough* NixOS *cough*) these won't be where you're expecting.